src/control/server/server.go

```diff
@@ -116,11 +114,13 @@ func Main() error {
 
 	mgmtpb.RegisterMgmtCtlServer(grpcServer, mgmtCtlSvc)
 
-	// Only provide IO/Agent communication if running as non-root user or if no
-	// non-root user has been specified in config (not recommended, will log error).
-	// If root, only provide gRPC mgmt control service for hardware provisioning.
-	if syscall.Getuid() != 0 || config.UserName == "" {
-		mgmtpb.RegisterMgmtSvcServer(grpcServer, newMgmtSvc(&config))
+	// If running as root and user name specified in config file, respawn proc.
+	respawn := syscall.Getuid() == 0 && config.UserName != ""
```

Comment:
suggestion: shouldRespawn is extremely clear and retains clarity regardless of how far away the variable is used from its declaration.